src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -35,9 +35,11 @@ internal abstract partial class AbstractTypeImportCompletionProvider : CommonCom
             SemanticModel semanticModel,
             CancellationToken cancellationToken);
 
+        protected abstract Task<bool> IsInImportsDirectiveAsync(Document document, int position, CancellationToken cancellationToken);
+
         // Telemetry shows that the average processing time with cache warmed up for 99th percentile is ~700ms.
         // Therefore we set the timeout to 1s to ensure it only applies to the case that cache is cold.
-        internal int TimeoutInMilliseconds => 1000;
+        internal static int TimeoutInMilliseconds { get; set; } = 1000;
```

Comment:
>static [](start = 17, length = 6)

What is the benefit of making it static?
Can we execute tests in parallel and break it?

Reply:
Other than that the PR looks good. But this one concerns me.

---
In reply to: [308491939](https://github.com/dotnet/roslyn/pull/37290#discussion_r308491939) [](ancestors = 308491939)

Reply:
That's a good point. The reason it's static is to make it easier to disable the timebox during tests (since it's not easy to get the provider object itself during tests, and I don't want to add an additional option just for test purpose)

There are 3 kinds of tests that might be affected by this.
1. completion provider tests that are specifically written for import completion. For those, this value is always set to -1 to avoid flakiness, so being static shouldn't be an issue.
2. completion command handler tests. Those are WfpFact/WpfTheory, which I believe can't be executed in parallel.
3. Integration tests. I don't think these can be executed in parallel either.

Reply:
You can do this through instance state that is passed in through a mef import.  i.e. something that is queried to get the delay.  In tests you pass in an instance that says 'infinite delay'.

Reply:
So something like this?
http://source.roslyn.io/#Roslyn.Services.Test.Utilities/TestExperimentationService.cs,13

Reply:
Statics are really really unpleasant.  It would be easy for this to lead to some *very* hard to debug flakeyness in tests.  I would really try to avoid it if possible.  Using an 'option' is fine with me as well.

Reply:
I totally agree, and will definitely fix this one way or another. I just want to avoid potential conflict this would cause with the expander PR. As I said above, I believe this doesn't affect tests for now. Once this is merged, I will update the expander PR and fix this there immediately (which will be merged as soon as our integration test machine is updated to 16.3 preview 1)

Reply:
Fixed.